core/src/test/java/nl/nn/adapterframework/senders/XsltSenderTest.java

```diff
@@ -91,4 +100,73 @@ public void testComplexXslt1Xpath() throws SenderException, TimeOutException, Co
 
 		assertEquals("2", result);
 	}
+
+	@Test
+	public void testDynamicStylesheet() throws ConfigurationException, IOException, PipeRunException, PipeStartException, SenderException {
+		sender.setStyleSheetName("/Xslt/dynamicStylesheet/wrongDummy.xsl");
+		sender.setStyleSheetNameSessionKey("stylesheetName");
+		sender.configure();
+		sender.open();
+
+		session = new PipeLineSessionBase();
+		session.put("stylesheetName", "/Xslt/dynamicStylesheet/correctDummy.xsl");
+
+		String input = TestFileUtils.getTestFile("/Xslt/dynamicStylesheet/in.xml");
+		log.debug("inputfile ["+input+"]");
+		String expected = TestFileUtils.getTestFile("/Xslt/dynamicStylesheet/out.txt");
+
+		ParameterResolutionContext prc = new ParameterResolutionContext(input, session);
+		assertEquals(expected, sender.sendMessage(null, input, prc));
```

Comment:
Hoe weet je nu dat het dynamische stylesheet gebruikt is, en niet de 'wrongDummy'?